src/date/expand-pattern.js

```diff
@@ -38,6 +39,89 @@ return function( options, cldr ) {
 		);
 	}
 
+	function getBestMatchPattern( path, skeleton ) {
+		var availableFormats, ratedFormats, format;
+
+		pattern = cldr.main([ path, skeleton ]);
+
+		if (skeleton && pattern === undefined) {
+			availableFormats = cldr.main([ path ]);
+			ratedFormats = [];
+
+			for (format in availableFormats) {
+				ratedFormats.push({
+					format: format,
+					pattern: availableFormats[format],
+					rate: compareFormats(skeleton, format)
+				});
+			}
+
+			ratedFormats.sort(function(a, b) {
+				return a.rate - b.rate;
+			});
+
+			if (ratedFormats.length) {
+				pattern = augmentFormat(skeleton, ratedFormats[0].pattern);
+			}
+		}
+
+		return pattern;
+	}
+
+	function compareFormats( a, b ) {
+		var distance, minLength, i;
+
+		distance = 1;
+		a = a.match(datePatternRe);
+		b = b.match(datePatternRe);
+		minLength = Math.min(a.length, b.length);
+
+		for (i = 0; i < minLength; i++) {
+			if (a[i].charAt(0) === b[i].charAt(0)) {
+				if (a[i].length === b[i].length) {
+					distance *= 0.25;
+				} else {
+					distance *= 0.75;
+				}
+			} else {
+				distance *= 1.25;
+			}
+		}
+
+		if (a.length === b.length) {
+			distance *= 0.5;
+		}
+
+		return distance;
+	}
+
+	function augmentFormat( requestedSkeleton, bestMatchFormat ) {
+		var originalBestMatchFormat, i, t, j, k, l;
+
+		originalBestMatchFormat = bestMatchFormat;
+		requestedSkeleton = requestedSkeleton.match(datePatternRe);
+		bestMatchFormat = bestMatchFormat.match(datePatternRe);
+
+		for (i = 0, l = bestMatchFormat.length; i < l; i++) {
```

Comment:
@jquery/globalize what's your thought about it? Is it worth for performance reasons or it's safe to compare to length on each iteration?